fix: #3043,#3044, #3048 Refactor several UI components and improve er…

Kasia9090 opened the pull request:
…ror handling

This commit includes several updates to improve user experience and enhance error handling. The toast.error method is replaced with toast.success in toastSuccess.tsx for success messages. The email composing component has been improved to handle potential errors correctly by using toastError.

Additions have been made to the getTimeline graphql query to fetch 'summary' and 'actionItems' fields which appears to be necessary for subsequent use.

Certain UI elements like SocialInput and UrlInput have been refactored for better hover functionality. The BCC option in ParticipantsSelectGroup component is commented out for undisclosed reasons and might be related to bug #3044.

Styling updates have been made in several places for a uniform and improved look and feel. For instance, box-shadow is being used instead of drop-shadow for better browser compatibility.

Also, to allow the user more time to read the toast messages, the autoClose time has been significantly increased in layout.tsx.

Lastly, debugging logs are added in the OrganizationTimeline.tsx file, presumably useful for the ongoing testing and debugging phase.

## Proposed changes

<!---Describe the big picture of your changes here.  If it fixes a bug or resolves a feature request, be sure to link that issue!--->

## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Additional context

Kasia9090 committed:
fix: #3043,#3044, #3048 Refactor several UI components and improve error handling

This commit includes several updates to improve user experience and enhance error handling. The toast.error method is replaced with toast.success in toastSuccess.tsx for success messages. The email composing component has been improved to handle potential errors correctly by using toastError.

Additions have been made to the getTimeline graphql query to fetch 'summary' and 'actionItems' fields which appears to be necessary for subsequent use.

Certain UI elements like SocialInput and UrlInput have been refactored for better hover functionality. The BCC option in ParticipantsSelectGroup component is commented out for undisclosed reasons and might be related to bug #3044.

Styling updates have been made in several places for a uniform and improved look and feel. For instance, box-shadow is being used instead of drop-shadow for better browser compatibility.

Also, to allow the user more time to read the toast messages, the autoClose time has been significantly increased in layout.tsx.

Lastly, debugging logs are added in the OrganizationTimeline.tsx file, presumably useful for the ongoing testing and debugging phase.

Kasia9090 committed:
Update email sending feedback

Updated the feedback messages that users get when they send an email. If the email is sent successfully, a toast success message will show up. If there are errors, the error reason and subject will be included in the toast error message. This would provide a better experience for users, as they will have a clearer idea about the result and possible issues.

Kasia9090 committed:
(chore): Build fix

steynviljoen67 commented on packages/apps/spaces/app/layout.tsx:
```diff
@@ -49,7 +49,7 @@ export default async function RootLayout({
               <ToastContainer
                 position='bottom-right'
-                autoClose={3000}
+                autoClose={3000000}
```
For the toast I would recommend a time of 8 seconds.

Kasia9090 replied:
done

Kasia9090 committed:
Update toast close timing and padding for email response

This commit changes the autoClose property for the ToastContainer in layout.tsx from 3000000 to 8000 to create a more responsive user interface. It also adjusts the padding on the left (pl) and right (pr) for the email response buttons in EmailResponseModeChangeButtons.tsx to enhance the visual appeal. These changes aim to improve the user's experience while interacting with notifications and the email response interface.
